lib/pages/search/search_contacts_and_chats_controller.dart

```diff
@@ -20,136 +17,96 @@ class SearchContactsAndChatsController {
-  static const limitContactsPerPage = 20;
-  static const debouncerIntervalInMilliseconds = 300;
-  static const minimumItemsListDisplay = 20;
+  static const int _limitPrefetchedRecentChats = 3;
+  static const _debouncerIntervalInMilliseconds = 300;
+
   final SearchRecentChatInteractor _searchRecentChatInteractor =
       getIt.get<SearchRecentChatInteractor>();
-  final _searchContactsInteractor = getIt.get<GetContactsInteractor>();
-  bool _isLoadingMore = false;
 
-  final recentAndContactsNotifier = ValueNotifier<Either<Failure, Success>>(
-    Right(SearchInitial()),
-  );
+  final ContactManager contactManger = getIt.get<ContactManager>();
+
+  final recentAndContactsNotifier = ValueNotifier<List<PresentationSearch>>([]);
   Debouncer<String>? _debouncer;
 
   MatrixLocalizations get _matrixLocalizations =>
       MatrixLocals(L10n.of(context)!);
+
   List<Room> get _rooms => Matrix.of(context).client.rooms;
 
-  void init() {
+  void init() async {
     _initializeDebouncer();
+    contactManger.initialSynchronizeContacts();
     fetchPreSearchChat();
   }
 
   void _initializeDebouncer() {
     _debouncer = Debouncer(
-      const Duration(milliseconds: debouncerIntervalInMilliseconds),
+      const Duration(milliseconds: _debouncerIntervalInMilliseconds),
       initialValue: '',
     );
 
     _debouncer?.values.listen((keyword) async {
       Logs().d(
         "SearchContactAndRecentChatController::_initializeDebouncer: searchKeyword: $keyword",
       );
-      searchChats(keyword: keyword);
+      _searchChatsFromLocal(keyword: keyword);
     });
   }
 
   void fetchPreSearchChat() {
     _searchRecentChatInteractor
         .execute(
-          keyword: '',
-          matrixLocalizations: _matrixLocalizations,
-          rooms: _rooms,
-          limit: limitPrefetchedRecentChats,
-        )
+      keyword: '',
+      matrixLocalizations: _matrixLocalizations,
+      rooms: _rooms,
+      limit: _limitPrefetchedRecentChats,
+    )
         .listen(
-          (event) => mapPreSearchChatToPresentation(event, isLoadMore: false),
-        );
+      (event) {
+        event.map((success) {
+          if (success is SearchRecentChatSuccess) {
+            recentAndContactsNotifier.value =
+                success.toPresentation().tomContacts;
+          }
+        });
+      },
+    );
   }
 
-  void searchChats({required String keyword}) {
+  void _searchChatsFromLocal({required String keyword}) {
     if (keyword.isEmpty) {
       return fetchPreSearchChat();
     }
+    final tomContacts = contactManger.tomContacts;
+    final tomPresentationSearchContacts = tomContacts
+        .expand((contact) => contact.toPresentationContacts())
+        .toList();
+    final recentChatPresentationSearchMatched = tomPresentationSearchContacts
+        .expand((contact) => contact.toPresentationSearch())
+        .where((contact) {
+      return contact.displayName!.toLowerCase().contains(keyword.toLowerCase());
+    }).toList();
```

Comment:
you should move it into Interactor

Reply:
1. The `Interactor` belongs to the model layer that only contains `Contact`. Because I don't use states, I only handle it here.
2. We  just only search local, not using `interactor` at the moments